sdks/python/apache_beam/pvalue.py

```diff
@@ -669,8 +669,8 @@ def __hash__(self):
     return hash(self.__dict__.items())
 
   def __eq__(self, other):
-    return type(self) == type(other) and all(
-        s == o for s, o in zip(self.__dict__.items(), other.__dict__.items()))
+    return (type(self) == type(other) and
+            self.__dict__.items() == other.__dict__.items())
```

Comment:
I think we'd like to check the order in which elements were added to the dictionary as per https://github.com/apache/beam/pull/14156 . Is that the case, @TheNeuralBit ?

If so, we could instead add another condition to make sure dicts has equal # of elements.
Would be nice to add a unit test for this case as well.

Reply:
I think `items` will take order into account. But it would be good to have a simple unit test that verifies `beam.Row(x=1, y=2) != beam.Row(y=2. x=1)`.

Reply:
I added a test test and went ahead with my suggestion. relying on `items()` alone was not sufficient.